designsafe/apps/api/publications/operations.py

```diff
@@ -254,3 +256,37 @@ def initilize_publication(publication, status='publishing', revision=None, revis
         # Refresh index so that search works in subsequent pipeline operations.
         IndexedPublication._index.refresh(using=es_client)
         return pub
+
+def clarivate_single_api(doi):
+
+    base_url = 'https://api.clarivate.com/apis/wos-starter/v1/documents'
+    apikey = os.environ.get('WOS_APIKEY')
+    
+    if not apikey:
+        return {"error": "Clarivate API key is missing"}
+
+    if not doi:
+        return {"error": "DOI is required"}
+
+    params = {'db': 'DRCI', 'q': f"DO={doi}"}
+    
+    try:
+        response = requests.get(
+            base_url,
+            headers={'X-Apikey': apikey},
+            params=params
+        )
+        response.raise_for_status()
+        rspdict = response.json()
+
+        citations = [
+            source['count'] for source in rspdict['hits'][0]['citations']
+            if source['db'] == 'WOK'
+        ][0]
+
+        return citations
+
+    except requests.exceptions.RequestException as e:
+        return {"error": str(e)}
+    except (KeyError, IndexError) as e:
+        return {"error": "Unexpected response format"}
```

Comment:
Can you please format this using the Black formatter? (it's available as a vscode extension)

I'm still trying to figure out how to enforce Python formatting at the CI level without retroactively modifying a million files and destroying git-blame

Reply:
changed to:
`except requests.exceptions.RequestException as e:
        return {
            "error": str(e),
        }
    except (KeyError, IndexError) as e:
        return {
            "error": "Unexpected response format",
        }`